docs/developer_guide.md

```diff
@@ -33,11 +33,104 @@ eval $(minikube docker-env)
 make build
 ```
 
-### Changing Types
+### 4. Changing Types
 
 If you're making a change to the `pkg/apis` package, please ensure you re-run
 following command for code regeneration.
 
 ```
 make codegen
 ```
+
+### Deploying and testing your changes on a cluster
```

Comment:
Thanks for adding all of these. Might be worth of mentioning local clusters like `k3d`, `kind` and `minikube` are supported for testing by running `make start`/`make image`, before giving an option of pushing the test image to any image registry (and mentioning ttl.sh is a convenient one). WDYT?